BUILD.md

```diff
@@ -0,0 +1,472 @@
+:balloon: Welcome to the Conduit development guide! :wave:
+
+This document will help you build Conduit from source. Depending on the
+components you are interested in building, these instructions assume you have
+the following tools already setup:
```

Comment:
I believe there are tools for auto-generating a Markdown TOC...

Reply:
I believe this file might be trying to do too much in a single place. I see three use cases for the write-up here:

1) I want to build Conduit from Master, either because I want to test the HEAD or because I've made a small change and want to make sure it works. Basically, all I need to know is what commands to run to get master deployed somewhere.
2) I want to contribute to Conduit and need to understand how the code is structured, what development practices to follow, and what each directory contains. I need to know about the modules, what is allowed or not, and how do I run a developer build and tests.
3) I want to understand Conduit's architecture, it's major components and how they relate to each other and to my application. I need to understand how the CLIs talks to the controller and how the controller talks to proxies.

I would suggest that we need a specific doc per item, something like `BUILD.md`, `DEVELOP.md`, and `ARCHITECTURE.md`.

Reply:
Agree this is a lot of info in one doc. I did consider breaking out an `ARCHITECTURE.md`, particularly for the dependency graphs. Having just consolidated info from several README's, I'd like to leave this info in a single doc for now. As specific sections increase in complexity, let's break them out at that point.